### DISCUSSION TITLE
How to remove directory icon from buffers section

### DISCUSSION THREAD

**Discussion by lsig** [nvim-lualine/lualine.nvim#1025] [Q&A]
I was wondering if it is possible to remove both the directory icon, and the space it takes on the lualine? When i set the symbol to "", It still takes up space. 
![Screenshot 2023-04-23 at 15 41 52](https://user-images.githubusercontent.com/89947902/233843487-76dfa2e6-4fdb-433e-92ce-8fb87f9436ef.jpg)
![Screenshot 2023-04-23 at 15 46 27](https://user-images.githubusercontent.com/89947902/233843488-ecaeae00-cc14-4530-82a0-b5aaf43c13ae.jpg)

**Comment by yutkat** [comment]
The space is inserted here.

https://github.com/nvim-lualine/lualine.nvim/blob/84ffb80e452d95e2c46fa29a98ea11a240f7843e/lua/lualine/components/buffers/buffer.lua#L55

So you can get a workaround with this hack.

```lua
			symbols = {
				directory = false,
			},
```

**Reply by yutkat** [reply]
@lsig  Have you solved this?

**Reply by lsig** [reply]
Hi, sorry for the slow reply.  I do not think that putting the directory to false fixes the issue. It seems as though the gap is still as wide. When I have to files in different nested directories, lualine still makes space for the directory icon, between the git changes and the list of buffers. 
![Screenshot 2023-04-27 at 10 38 41](https://user-images.githubusercontent.com/89947902/234807626-79e08b18-e06e-4235-a737-f31a934ad8db.jpg)

![Screenshot 2023-04-27 at 10 39 48](https://user-images.githubusercontent.com/89947902/234807935-9216c90b-8c2a-4fd9-834c-ec83907cb963.jpg)

This is how it looks when directory is not false:

![Screenshot 2023-04-27 at 10 58 48](https://user-images.githubusercontent.com/89947902/234813091-e950bbbc-5d76-4685-bfe2-c4e2ea006c40.jpg)

**Reply by lsig** [reply]
This is my current lualine config: https://github.com/lsig/.nvim/blob/main/after/plugin/lualine.lua

**Reply by yutkat** [reply]
`:lua vim.fn.setcellwidths({{0xe0b0, 0xe0b3, 1}})`

**Reply by lsig** [reply]
This does not seem to change anything when I run it as a command or put it above me lualine config (without the lua)

**Reply by yutkat** [reply]
I think I've tried it with minimal configs and it doesn't seem to show any wasted space.

before:
![swappy-20230428-003012](https://user-images.githubusercontent.com/8683947/234911692-3bc0227a-f4be-4228-9139-c42c5d9fc19c.png)

after:
![swappy-20230428-002913](https://user-images.githubusercontent.com/8683947/234911388-69c7e510-d6b9-460d-99ae-6df11a2d066b.png)

minimal config is here.

```lua
vim.cmd([[syntax enable]])
vim.cmd([[filetype plugin indent on]])

local lazypath = vim.fn.stdpath("data") .. "/lazy/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
	vim.fn.system({
		"git",
		"clone",
		"--filter=blob:none",
		"https://github.com/folke/lazy.nvim.git",
		"--branch=stable", -- latest stable release
		lazypath,
	})
end
vim.opt.rtp:prepend(lazypath)
vim.cmd[[colorscheme desert]]
vim.o.tabstop = 2
vim.o.mouse = false

require("lazy").setup({
	{ "nvim-lua/plenary.nvim" },
	{ "MunifTanjim/nui.nvim" },
	{ "neovim/nvim-lspconfig" },
	{
		"williamboman/mason.nvim",
		event = "BufReadPre",
		config = function()
			require("mason").setup()
		end,
	},
	{
		"williamboman/mason-lspconfig.nvim",
		event = "BufReadPre",
		config = function()
			require("mason-lspconfig").setup()
			local lspconfig = require("lspconfig")
			require("mason-lspconfig").setup_handlers({
				function(server_name)
					lspconfig[server_name].setup({})
				end,
			})
		end,
	},
	{"nvim-tree/nvim-web-devicons"},
	{
		"nvim-lualine/lualine.nvim",
		event = "BufReadPre",
		config = function()
			require("lualine").setup({
				options = {
					icons_enabled = true,
					component_separators = { left = "", right = "" },
					section_separators = { left = "", right = "" },
					disabled_filetypes = {
						statusline = {},
						winbar = {},
					},
					ignore_focus = {},
					always_divide_middle = true,
					globalstatus = false,
					refresh = {
						statusline = 1000,
						tabline = 1000,
						winbar = 1000,
					},
				},
				sections = {
					lualine_a = { "mode" },
					lualine_b = { "branch", "diff", "diagnostics" },
					lualine_c = {
						{ "buffers", symbols = {
							alternate_file = "# ",
							--directory = false,
						} },
					},
					lualine_x = { "encoding", "filetype" },
					lualine_y = { "progress" },
					lualine_z = { "location" },
				},
				inactive_sections = {
					lualine_a = {},
					lualine_b = {},
					lualine_c = { "filename" },
					lualine_x = { "location" },
					lualine_y = {},
					lualine_z = {},
				},
				tabline = {},
				winbar = {},
				inactive_winbar = {},
				extensions = { "fugitive", "fzf" },
			})
		end,
	},
}, {
	defaults = {
		lazy = true,
	},
})
```
